commands/system/help.js

```diff
@@ -0,0 +1,326 @@
+      const output = sorted.filter(c => !(level < 10 && c.help.category === "Owner") || !(c.help.category === "NSFW" && !message.channel.nsfw)).map((c) => {
+        const cat = c.help.cat;
+        if (currentCategory !== cat && !type) {
+          currentCategory = cat;
+          return `\n\`${message.settings.prefix}help ${cat.toLowerCase()}\` | Shows ${cat} commands`;
+        }
+      }).join("");
+      embed.setDescription(description)
+        .addField("Categories", output);
+    } else {
+      sorted.forEach((c) => {
+        if (c.help.cat.toLowerCase() === type.toLowerCase()) {
+          n = n + 1;
+        }
+      });
+
+      var output = "";
+      num = 0;
+      const pg = parseInt(page) && parseInt(page) <= Math.ceil(n / perpage) ? parseInt(page) : 1;
+      for (const c of sorted.values()) {
+        if (c.help.cat.toLowerCase() === type.toLowerCase()) {
+          if (c.help.cat === "NSFW" && !message.channel.nsfw) return;
+          if (num < perpage * pg && num > perpage * pg - (perpage + 1)) {
+            if (level < this.client.levelCache[c.conf.permLevel]) return;
+            output = output + `\n\`${message.settings.prefix + c.help.name}\` | ${c.help.description.length > 80 ? `${c.help.description.slice(0, 80)}...` : c.help.description}`;
+          }
+          num = num + 1;
+        }
+      }
+
+      if (num) {
+        finalpage = Math.ceil(num / perpage);
+        embed.setTitle(`Page ${page}/${Math.ceil(num / perpage)}`)
+          .setDescription(`A list of commands in the ${type} category.\n(Total of ${num} commands in this category)\n\nTo get help on a specific command do \`${message.settings.prefix}help <command>\`\n\n${num > 10 && pg === 1 ? `To view more commands do\` ${message.settings.prefix}help <category> 2\`` : ""}`)
+          .addField("Commands", output);
+      }
+    }
+
+    if (this.client.commands.has(type) || this.client.commands.some(command => command.conf.aliases.includes(type))) {
+      const cm = this.client.commands.get(type) || this.client.commands.get(this.client.aliases.get(type));
+      if (level < this.client.levelCache[cm.conf.permLevel]) return;
+      embed.setTitle(cm.help.name)
+        .addField("Command description", cm.help.description)
+        .addField("Command usage", `\`${cm.help.usage}\``)
+        .addField("Command aliases", cm.conf.aliases.length === 0 ? "None" : cm.conf.aliases.join(", "));
+    }
+
+    const msg2 = await message.channel.send(embed);
+    if (!message.guild.me.hasPermission(["MANAGE_MESSAGES"])) {
+      await message.channel.send("I don\"t have permission to remove reactions, please do this manually.");
+    }
+    if (msg2.embeds[0].title) {
+      await msg2.react("⏮");
+      await msg2.react("◀");
+      await msg2.react("⏹");
+      await msg2.react("▶");
+      await msg2.react("⏭");
+      await msg2.react("🔢");
+    }
+
+    const nextpage = msg2.createReactionCollector(
```

Comment:
Just, no. Unify all collectors in one. This one uses +7 times more RAM than what it should.

Reply:
But this command will barely use ram, maybe maximum 1MB if ram, and considering York has cool downs this can’t be abused, so I don’t see the issue of this

Reply:
tbh still what kyra said